tools/get_deps.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+
+PARSE_PKGS=./tools/parse-pkgs.sh
+
+get_deps() {
+    deps=$(sed '/^#/d' < "$1" | grep "FROM lfedge" | awk '{print $2}' | cut -d":" -f1 | sed -e "/.*eve-alpine.*/d" -e "/.*eve-kernel.*/d")
+    for d in $deps; do
+        #shellcheck disable=SC2001
+        dockfile=$(echo "$d" | sed 's#^lfedge/\(eve-\|\)\(.*\)#pkg/\2/Dockerfile#')
+        if [ -e "$dockfile" ]; then
+            get_deps "$dockfile"
+        else
+            echo "ERROR: $dockfile not found."
+            break
+        fi
+    done
+    # There is no more dependencies, just print the name of the package
+    dirname "$1"
+}
+
+print_head() {
+    if [ "$1" = "IMG" ]; then
+        cat << EOF
+digraph unix {
+  rankdir="LR";
+EOF
+    else
+        cat << EOF
+#
+# This file was generated by EVE's build system. DO NOT EDIT.
+#
+EOF
+    fi
+}
+
+print_single_dep() {
+    if [ "$1" = "IMG" ]; then
+        echo "\"$pkg\";"
+    fi
+}
+
+print_dep() {
+    if [ "$1" = "IMG" ]; then
+        echo "\"$2\" -> \"$3\";"
+    else
+        echo "$2: $3"
+    fi
+}
+
+print_tail() {
+    if [ "$1" = "IMG" ]; then
+        echo "}"
+    fi
+}
+
+gen_dep_graph() {
+    # Scan all directories of pkg/
+    print_head "$1"
+    PKGS=$(ls -d pkg/*)
+    for pkg in $PKGS; do
+        print_single_dep "$1" "$pkg"
+        DOCKFILE="$pkg/Dockerfile"
+        REMDOCKFILE=false
+
+        if [ ! -e "$DOCKFILE" ]; then
+            if [ -e "$pkg/Dockerfile.in" ]; then
+                DOCKFILE=$(mktemp)
+                $PARSE_PKGS "$pkg/Dockerfile.in" > "$DOCKFILE"
+                REMDOCKFILE=true
+            else
+                continue
+            fi
+        fi
+
+        deps=$(get_deps "$DOCKFILE")
+        if [ "$REMDOCKFILE" = "true" ]; then
+            rm "$DOCKFILE"
+        fi
+
+        for d in $deps; do
+            if [ "$d" = "$pkg" ] || [ "$d" = "/tmp" ]; then
+                continue
+            else
+                print_dep "$1" "$pkg" "$d"
+            fi
+        done
+    done
+
+    # Scan rootfs dependencies
+    if [ "$DROOTFS" = "true" ]; then
+        for x in images/out/*.yml; do
```

Comment:
`images/out/*.yml` might be empty and then you get an error message like:
```
tools/get_deps.sh: line 92: images/out/*.yml: No such file or directory
```

Reply:
Thanks for pointing that out, I will fix it...

Reply:
fixed!